File: solutions/sonnet-1/sonnet/typescript/transpose.ts
const input = await Bun.stdin.text();
const tokens = input.split(/\s+/).filter((s) => s.length > 0).map(Number);

let idx = 0;
const R = tokens[idx++];
const C = tokens[idx++];

const matrix: number[][] = [];
for (let r = 0; r < R; r++) {
  const row: number[] = [];
  for (let c = 0; c < C; c++) {
    row.push(tokens[idx++]);
  }
  matrix.push(row);
}

const lines: string[] = [];
for (let c = 0; c < C; c++) {
  const row: number[] = [];
  for (let r = 0; r < R; r++) {
    row.push(matrix[r][c]);
  }
  lines.push(row.join(" "));
}

console.log(lines.join("\n"));

File: solutions/sonnet-1/sonnet/perl/transpose.pl
use strict;
use warnings;

my $first = <STDIN>;
my ($r, $c) = split ' ', $first;

my @matrix;
for my $i (0 .. $r - 1) {
    my $line = <STDIN>;
    chomp $line;
    my @vals = split ' ', $line;
    push @matrix, \@vals;
}

for my $j (0 .. $c - 1) {
    my @col;
    for my $i (0 .. $r - 1) {
        push @col, $matrix[$i][$j];
    }
    print join(' ', @col), "\n";
}
